### PR TITLE
fix: plugin doesn't show welcome view (#12458)

### PR DESCRIPTION
**Pull request by helloworldmygithub**
#### What it does

Closes: #12458 
The pull-request checks the view data provider after the start of a plugin to show the plugin's welcome view.

#### How to test

1. start theia
2. download the plugin installation package postcode 1.3.9）
 3. install the plugin using `isntall from vsix...`

_Before_:
image

_After_:
image

#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

File: packages/plugin-ext/src/hosted/browser/hosted-plugin.ts [hunk 1]
```diff
@@ -303,6 +303,9 @@ export class HostedPluginSupport {
         }
         await this.startPlugins(contributionsByHost, toDisconnect);
 
+        // check data provider after start plugin to show welcom view
```
Comment (vince-fugnitto) [inline packages/plugin-ext/src/hosted/browser/hosted-plugin.ts:306]: minor: 
```suggestion
        // check data provider after plugin start to show welcome view
```

File: packages/plugin-ext/src/main/browser/view/plugin-view-registry.ts [hunk 2]
```diff
@@ -780,19 +780,48 @@ export class PluginViewRegistry implements FrontendApplicationContribution {
...
+                            }));
+                            const unsubscribe = () => toDisposeOnDidExpandView.dispose();
+                            view.disposed.connect(unsubscribe);
+                            toDisposeOnDidExpandView.push(Disposable.create(() => view.disposed.disconnect(unsubscribe)));
+                        }
+                    }
+                });
+            }
+        }
+    }
```
Comment (vince-fugnitto) [inline packages/plugin-ext/src/main/browser/view/plugin-view-registry.ts:805]: This code duplicates a lot of what `registerViewDataProvider` already does, I don't think it is the proper solution as is.

Comment (helloworldmygithub) [reply packages/plugin-ext/src/main/browser/view/plugin-view-registry.ts:805]: May I ask if the meaning here is that I should try to reuse code? Or should we solve the problem in a different way

Comment (vince-fugnitto) [reply packages/plugin-ext/src/main/browser/view/plugin-view-registry.ts:805]: Sorry, I mean that we should not have duplication, and it looks like the current solution duplicates a lot of what `registerViewDataProvider` already does without any clear reason as to why it does. We should find a solution which does not duplicate the code, either by seeing what can be reused once we better understand the problem, or a completely different solution if necessary.

Comment (helloworldmygithub) [reply packages/plugin-ext/src/main/browser/view/plugin-view-registry.ts]: Okay, I understand. In Theia，widgets should be opened only when view data providers are registered, but some vscode plugins only have some welcomeViews and do not register a viewDataProvider. So I want to check if the plugin view where the welcome view is located has been registered a viewDataProvider after the plugin is launched, and if not, give a default

Comment (helloworldmygithub) [reply packages/plugin-ext/src/main/browser/view/plugin-view-registry.ts]: Thank you for your previous suggestion, but I can't find a simpler solution. I will create a new PR to remove the commit messages and wait for other results

**Pull request closed without merging**